src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/IngestionFlowFileAchiverService.java

```diff
@@ -0,0 +1,69 @@
+package it.gov.pagopa.payhub.activities.service.ingestionflow;
+
+import it.gov.pagopa.payhub.activities.service.ZipFileService;
+import it.gov.pagopa.payhub.activities.util.AESUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+/**
+ * Service class responsible for handling archiving ingestion files,
+ * including zipping and ecryption files.
+ */
+@Lazy
+@Service
+public class IngestionFlowFileAchiverService {
+	/**
+	 * The target directory used for storing processed files.
+	 */
+	private final String targetDirectory;
+
+	/**
+	 * The password used for encrypting files.
+	 */
+	private final String dataCipherPsw;
+
+	private final ZipFileService zipFileService;
+
+	public IngestionFlowFileAchiverService(@Value("${stored-dir:/processed/}") String targetDirectory,
```

Comment:
the target directory should be on the shared folder as the original file.
should the target directory be a subdirectory of the original position?

Reply:
We will use the original file path, the target dir will be a subfolder